src/org/rascalmpl/repl/BaseREPL.java

```diff
@@ -127,7 +127,12 @@ public void run() throws IOException {
                         break;
                     }
                     running.set(true);
-                    handleInput(line);
+                    if (line.equals("\n")) {
```

Comment:
I'm sorry, but in my mind, this should be at the same level as the line parser. What a ParametricREPL considers a canceled input is not what we want to deal with, since that also depends on the parser of the input.

I think the double line handling should be done at the `RascalReplServices` level.